reactioncommerce/reaction
# fix: Can't re-send invite to existing non-activated user

Resolves #5176  
Impact: **major**  
Type: **bugfix**

## Issue
After sending a shop manager invite to a non-registered e-mail, a new user is created for them with a token to be used to set their password. An e-mail invite is then sent with a URL containing this token.

However, if you want to re-send an invitation to this same person, the `accounts/inviteShopMember` will see an existing account for this e-mail, not checking whether it was activated or not. This means that no new "set password" token will be sent, and the user will be promoted to shop manager (which it already was), as if it was an already active account.

## Solution
In the `accounts/inviteShopMember` logic, if an account already exists for the e-mail address but isn't activated yet, generate a new token and send a new invite.

Also, add some confirmation dialogs in case an e-mail already exists, whether it's activated or not.

## Breaking changes
None.

## Testing
1. In the `Accounts` page of the operator, invite an e-mail address that's already registered and activated.
2. See a dialog asking you if you want to promote this existing user to shop manager.
3. Invite an e-mail that's never been used for an account before.
4. Receive invite on that e-mail. Delete it "by mistake".
5. Re-invite this same e-mail again in an attempt to send a new invite.
6. See a dialog asking you if you want to send a new invitation. Click yes.
7. Receive a new invitation with a new token.
8. Click on the button in the e-mail, see the "set password" form.
